### PR TITLE
added  Recommendations , api fetch and database check

### PR DESCRIPTION
**Pull request by divyanshsingh101**
issue name - Fetch details about the movie using an API #3
## Description

i have added the recommendation of the movie based on the characters entered in the title field  
if only title is entered then it will fetch data from the api else we have to enter the data from our own 
it checks if the title is present in the database before adding 

there we some bugs i fixed like it add it was redirecting to user/add_movies/movie_list  but it should redirect to user/movie_list 
there was no home page route at first 
user models needed to be migrated (i have updated the readme )

## Related Issue

Fixes # (issue)

## Type of change

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x] This change requires a documentation update

## Screenshots (if appropriate):
Screenshot 2024-10-09 112818
Screenshot 2024-10-08 235815

## Checklist:

- [x ] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [ x] My code follows the code style of this project.
- [ x] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings
- [x ] My change requires a change to the documentation.
- [x ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by eleensmathew** [comment]
delete .env file from commit and add .env in gitignore

File: README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # MovieScreen
 
-A basic Django website for first-time learners.
+A basic Django website for first-time learners.<br>
+enter api key in .env -by divyansh :)
```
Comment (eleensmathew) [inline README.md]: No need for this comment here

File: movie/urls.py [hunk 2]
```diff
@@ -18,10 +18,10 @@
...
 from django.conf.urls.static import static
 from django.conf import settings
-
+from user.views import home
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('user/', include(('user.urls', 'user'), namespace='user')),
-    
+    path('', home, name='home'),
```
Comment (eleensmathew) [inline movie/urls.py]: why are the urls changed? the url for home should be /user/

File: user/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Generated by Django 5.1.1 on 2024-10-08 14:31
+
```
Comment (eleensmathew) [inline user/migrations/0001_initial.py]: migrations foldre should not be commited. Pls delete and add this folder to gitignore.

File: movie/urls.py (hunk 2 above)
Comment (divyanshsingh101) [reply movie/urls.py]: i thought the "/" url is missing

File: user/migrations/0001_initial.py (hunk 3 above)
Comment (eleensmathew) [reply user/migrations/0001_initial.py]: Same goes for pycache files

Comment (divyanshsingh101) [reply user/migrations/0001_initial.py]: ok ma'am do i need to make a new pr ?

File: user/templates/movie_list.html [hunk 4]
```diff
@@ -8,14 +8,18 @@
...
-                <li>{{ movie.description }}</li>
-                <li><img src="{{ movie.poster.url }}" width="100"></li>
-            </div>
-        {% empty %}
-            <li>No movies available.</li>
-        {% endfor %}
+    <div style="padding: 10px;">
+        <li>{{ movie.title }}</li>
+        <li>{{ movie.description }}</li>
+        {% if movie.poster %}
```
Comment (eleensmathew) [inline user/templates/movie_list.html]: movie poster must always be there so remove this if condition

File: user/urls.py [hunk 5]
```diff
@@ -2,14 +2,18 @@
...
+    path('movie_list/',MovieListView.as_view(), name='movie_list'),
     path('add_movie/', add_movie, name='add_movie'),
-    path('movie_list/', movie_list, name='movie_list'),
     path('add_show/', AddShowView.as_view(), name='AddShowView'),
+     path('autocomplete/', MovieAutocomplete.as_view(), name='movie_autocomplete'),
     
     
-]
\ No newline at end of file
+]
```
Comment (eleensmathew) [inline user/urls.py:15]: [nit] remove extra lines

File: movie/urls.py (hunk 2 above)
Comment (divyanshsingh101) [reply movie/urls.py]: - first it will try to access only http://127.0.0.1:8000/ route

File: user/views.py [hunk 6]
```diff
@@ -1,30 +1,115 @@
...
-    return render(request, 'shows.html', {'shows': upcoming_shows})
+    shows = Show.objects.all().order_by('-time')
+    return render(request, 'shows.html', {'shows': shows})
+
+
+def home(request):
+    return render(request, 'base.html')
 
+
+class MovieAutocomplete(View):
```
Comment (eleensmathew) [inline user/views.py:21]: Can you explain this logic and add comments for this code

File: user/views.py [hunk 7]
```diff
@@ -42,3 +127,8 @@ def post(self, request, *args, **kwargs):
             form.save()
             return redirect('user:shows')
         return render(request, self.template_name, {'form': form})
+
```
Comment (eleensmathew) [inline user/views.py:113]: there is already a movie_list view so why is this added?

File: user/views.py (hunk 6 above)
Comment (divyanshsingh101) [reply user/views.py:21]: i am taking the query which is title name i should greater then 2 char then we are asking the api for the movies which contains the name and we are converting that data into json and showing it in dataset

File: movie/urls.py (hunk 2 above)
Comment (eleensmathew) [reply movie/urls.py]: i want the http://127.0.0.1:8000/user/shows/ to be valid and not http://127.0.0.1:8000/

File: user/migrations/0001_initial.py (hunk 3 above)
Comment (eleensmathew) [reply user/migrations/0001_initial.py]: no just make the new commits in the same pr

File: user/views.py (hunk 7 above)
Comment (divyanshsingh101) [reply user/views.py:113]: only due to naming view are conventionally named as MovieListView end with you i is not necessary though

File: user/templates/movie_list.html (hunk 4 above)
Comment (divyanshsingh101) [reply user/templates/movie_list.html]: if we are entering all the data by ourself  and doesn't give the poster then it handles the error

File: user/views.py (hunk 7 above)
Comment (eleensmathew) [reply user/views.py:113]: you have made a class based view while the older one is a function based view. They are both the same though so ig you can remove this.

File: movie/urls.py (hunk 2 above)
Comment (divyanshsingh101) [reply movie/urls.py]: should i redirect it to second route but will then also hit http://127.0.0.1:8000/ route then get redirected

File: user/templates/movie_list.html (hunk 4 above)
Comment (eleensmathew) [reply user/templates/movie_list.html]: make the posters a required field

File: movie/urls.py (hunk 2 above)
Comment (eleensmathew) [reply movie/urls.py]: no leave the url logic intact

Comment (divyanshsingh101) [reply movie/urls.py]: means i don't need to change this ??

Comment (eleensmathew) [reply movie/urls.py]: remove the changes you made i mean

File: user/templates/movie_list.html (hunk 4 above)
Comment (divyanshsingh101) [reply user/templates/movie_list.html]: if i will do so then i am populating the backend but the frontend will not allow me to submit the form as poster is required

File: movie/urls.py (hunk 2 above)
Comment (divyanshsingh101) [reply movie/urls.py]: so what will be there when the http://127.0.0.1:8000/  gets hit ?
do i need to redirect ?

File: user/templates/movie_list.html (hunk 4 above)
Comment (eleensmathew) [reply user/templates/movie_list.html]: Yes it should be like that

File: movie/urls.py (hunk 2 above)
Comment (eleensmathew) [reply movie/urls.py]: no thats a separate issue. No need to worry about that

File: user/templates/movie_list.html (hunk 4 above)
Comment (divyanshsingh101) [reply user/templates/movie_list.html]: i am only taking title from the frontend for api fetch so how can i make poster required at client side whereas in server side if the data comming from api then it will have poster     if i make poster required then i need to make two forms one only contains title and other that contain all the fields where i can make all fields required

Comment (eleensmathew) [reply user/templates/movie_list.html]: Ok interesting. One thing you can do is fetch the details of the movie only from the api. So if a weird movie title like Eleena is entered we should'nt save that in the database. Or what you can do is if the image isn't entered save a default image as the poster.

Comment (divyanshsingh101) [reply user/templates/movie_list.html]: it already has the feature that if title not found and do not entered any other field then it discard the entry   so i just need to remove the other fields from frontend?

Comment (eleensmathew) [reply user/templates/movie_list.html]: yes it can be removed

**Comment by divyanshsingh101** [comment]
ma'am i have updated the pull request

File: user/migrations/0001_initial.py (hunk 3 above)
Comment (eleensmathew) [reply user/migrations/0001_initial.py]: Pls delete this file and add to gitignore

**Comment by eleensmathew** [comment]
Can you provide more info on what api you have used?

**Comment by divyanshsingh101** [comment]
[OMDb API - The Open Movie Database](https://www.omdbapi.com/)

**Comment by divyanshsingh101** [comment]
it give data in json format like description , poster etc

**Comment by eleensmathew** [comment]
How have you test autocomplete? Did you check this functionality?
This is what I can see 
image

**Comment by divyanshsingh101** [comment]
so this require a query consisting of title name  is in not for general hitting without any query

**Comment by divyanshsingh101** [comment]
we can protect it from general api calls with auth

File: user/views.py [hunk 8]
```diff
@@ -1,28 +1,107 @@
...
+            
+                return JsonResponse([], safe=False)
+        return JsonResponse([], safe=False)
+
 
+# it will fetch data from api if only title field is  provided 
 def add_movie(request):
     if request.method == 'POST':
         title = request.POST.get('title')
         description = request.POST.get('description')
```
Comment (eleensmathew) [inline user/views.py]: Since you are not getting the description or poster from front end you can remove these lines

**Comment by eleensmathew** [comment]
Just make the final change and I will merge your pr

**Comment by divyanshsingh101** [comment]
ma'am i have made changes
